File: 3785.minimum-swaps-to-avoid-forbidden-values.java
#
# @lc app=leetcode id=3785 lang=java
#
# [3785] Minimum Swaps to Avoid Forbidden Values
#
# @lc code=start
class Solution {
public int minSwaps(int[] nums, int[] forbidden) {
    int n = nums.length;
    Map<Integer, Integer> map = new HashMap<>();
    for (int i = 0; i < n; i++) {
        if (nums[i] != forbidden[i]) {
            map.put(nums[i], i);
        } else {
            boolean swapped = false;
            for (int j = i + 1; j < n; j++) {
                if (nums[j] != forbidden[i] && nums[j] != forbidden[j]) {
                    int temp = nums[i];
                    nums[i] = nums[j];
                    nums[j] = temp;
                    swapped = true;
                    break;
                }
            }
            if (!swapped) return -1; // No valid swap found to separate from forbidden value. 
        }
    }
    return map.size(); // Count of successful swaps made. 
}
}
# @lc code=end

File: 3785.minimum-swaps-to-avoid-forbidden-values.go
#
# @lc app=leetcode id=3785 lang=golang
#
# [3785] Minimum Swaps to Avoid Forbidden Values
#

# @lc code=start
func minSwaps(nums []int, forbidden []int) int {
    n := len(nums)
    swapCount := 0
    conflictIndices := make([]int, 0)
    nonConflictIndices := make(map[int]bool)
    
    for i := 0; i < n; i++ {
        if nums[i] == forbidden[i] {
            conflictIndices = append(conflictIndices, i)
        } else {
            nonConflictIndices[i] = true
        }
    }
    
    if len(conflictIndices) == 0 {
        return 0
    }

    for _, ci := range conflictIndices {
        foundSwap := false
        for ai := range nonConflictIndices {
            if nums[ai] != forbidden[ci] && nums[ci] != forbidden[ai] {
                nums[ci], nums[ai] = nums[ai], nums[ci]
                delete(nonConflictIndices, ai)
                swapCount++
                foundSwap = true
                break
            }
        }
        if !foundSwap {
            return -1
        }
    }
    return swapCount
}
# @lc code=end